code/game/objects/items/camping.dm

```diff
@@ -0,0 +1,383 @@
+	var/edge1
+	var/edge2
+	var/axis
+	if(dir & (NORTH | SOUTH))
+		edge1 = x1
+		edge2 = x2
+		axis = canvas.x
+	else
+		edge1 = y1
+		edge2 = y2
+		axis = canvas.y
+	if(axis == edge1 || axis == edge2)
+		return "edge"
+	else if(axis == Mean(edge1, edge2))
+		return "mid"
+	return "norm"
+
+/obj/item/tent
+	name = "expedition tent"
+	desc = "A rolled up tent, ready to be assembled to make a base camp, shelter, or just a cozy place to chat."
+	desc_info = "Drag this to yourself to begin assembly. This will take some time, in 4 stages. Others can start working on the other stages by dragging it to themselves as well."
+	icon = 'icons/obj/item/camping.dmi'
+	icon_state = "tent"
+	item_state = "tent"
+	contained_sprite = TRUE
+	w_class = ITEMSIZE_LARGE
+	color = "#58a178"
+	var/width = 2
+	var/length = 3
+	var/decal
+
+	var/datum/large_structure/tent/my_tent
+
+/obj/item/tent/Initialize()
+	. = ..()
+	w_class = min(ceil(width * length / 1.5), ITEMSIZE_IMMENSE) // 2x2 = ITEMSIZE_NORMAL
+	desc += "\nThis one is [width] x [length] in size."
+
+/obj/item/tent/Destroy()
+	if(my_tent)
+		my_tent.source_item = null
+		if(!my_tent.grouped_structures)
+			QDEL_NULL(my_tent)
+	return ..()
+
+/obj/item/tent/MouseDrop(over_object, src_location, over_location)
+	. = ..()
+	if(use_check(usr) || !Adjacent(usr))
+		return
+	var/turf/T = get_turf(src)
+	if(istype(T))
+		deploy_tent(T, usr)
+
+/obj/item/tent/proc/deploy_tent(var/turf/target, var/mob/user)
+	if(my_tent)
+		if(my_tent.origin == get_turf(src)) //Not moved
+			my_tent.assemble(1 SECOND, user)
+			return
+		else
+			QDEL_NULL(my_tent)
+
+	var/deploy_dir = get_compass_dir(user,target)
+	if(target == get_turf(user))
+		deploy_dir = user.dir
+
+	my_tent = new /datum/large_structure/tent(src)
+	setup_my_tent(deploy_dir, target)
+
+	my_tent.assemble(1 SECOND, user)
+
+/obj/item/tent/proc/setup_my_tent(var/deploy_dir, var/turf/target)
+	my_tent.name = name
+	my_tent.color = color
+	my_tent.decal = decal
+	my_tent.dir = deploy_dir
+	my_tent.z1 = target.z
+	my_tent.z2 = target.z
+	my_tent.source_item_type = type
+	my_tent.source_item = src
+	my_tent.origin = get_turf(src)
+
+	if(deploy_dir & NORTH)
+		my_tent.x1 = target.x - floor((width-1)/2)
+		my_tent.x2 = target.x + ceil((width-1)/2)
+		my_tent.y1 = target.y
+		my_tent.y2 = target.y + (length-1)
+	else if(deploy_dir & SOUTH)
+		my_tent.x1 = target.x - ceil((width-1)/2)
+		my_tent.x2 = target.x + floor((width-1)/2)
+		my_tent.y1 = target.y - (length-1)
+		my_tent.y2 = target.y
+	else if(deploy_dir & EAST)
+		my_tent.x1 = target.x
+		my_tent.x2 = target.x + (length-1)
+		my_tent.y1 = target.y - ceil((width-1)/2)
+		my_tent.y2 = target.y + floor((width-1)/2)
+	else
+		my_tent.x1 = target.x - (length-1)
+		my_tent.x2 = target.x
+		my_tent.y1 = target.y - floor((width-1)/2)
+		my_tent.y2 = target.y + ceil((width-1)/2)
+
+/obj/item/tent/big
+	name = "scc base camp tent"
+	color = "#2e3763"
+	width = 3
+	length = 4
+	decal = "scc"
```

Comment:
make `/obj/item/tent/big` generic without the SCC logo
and add `/obj/item/tent/big/scc` with SCC logo